packages/snaps-cli/src/commands/build/build.ts

```diff
@@ -1,76 +1,59 @@
 import { isFile } from '@metamask/snaps-utils/node';
 import { assert } from '@metamask/utils';
-import { resolve as pathResolve } from 'path';
 
 import { build } from './implementation';
 import { getBundleAnalyzerPort } from './utils';
 import type { ProcessedConfig } from '../../config';
 import { CommandError } from '../../errors';
 import type { Steps } from '../../utils';
-import { success, executeSteps, info } from '../../utils';
-import { evaluate } from '../eval';
+import { success, executeSteps } from '../../utils';
 
 export type BuildContext = {
   analyze: boolean;
   build: boolean;
   config: ProcessedConfig;
+  exports?: string[];
   port?: number;
 };
 
 export const steps: Steps<BuildContext> = [
   {
     name: 'Checking the input file.',
-    condition: ({ build }) => build,
+    condition: ({ build: enableBuild }) => enableBuild,
     task: async ({ config }) => {
       const { input } = config;
 
       if (!(await isFile(input))) {
         throw new CommandError(
-          `Input file not found: "${input}". Make sure that the "input" field in your snap config is correct.`,
+          `Input file not found: "${input}". Make sure that the "input" field in your Snap config is correct.`,
         );
       }
     },
   },
   {
-    name: 'Building the snap bundle.',
-    condition: ({ build }) => build,
-    task: async ({ analyze, build: enableBuild, config, spinner }) => {
+    name: 'Building the Snap bundle.',
+    condition: ({ build: enableBuild }) => enableBuild,
+    task: async (context) => {
+      const { analyze, config, spinner } = context;
+
       // We don't evaluate the bundle here, because it's done in a separate
       // step.
       const compiler = await build(config, {
         analyze,
-        evaluate: false,
+        evaluate: config.evaluate,
         spinner,
       });
 
       if (analyze) {
         return {
-          analyze,
-          build: enableBuild,
-          config,
-          spinner,
+          ...context,
           port: await getBundleAnalyzerPort(compiler),
         };
       }
 
       return undefined;
     },
   },
-  {
-    name: 'Evaluating the snap bundle.',
```

Comment:
This is now handled by the Webpack plugin.

Reply:
How does `watch` work? I guess we may want to disable some of the permission validation during watch so it doesn't remove stuff you are presently adding?

Reply:
Good point, `watch` currently works the same as build. We don't have an easy way to prevent fixing certain validation rules, so for now I can just revert 21290d8.

Reply:
That would be a regression though right? Eval works in watch mode today does it not?

Reply:
We don't eval in watch mode currently.

Reply:
I think long term we can enable eval in watch mode and support partially fixing the manifest, e.g., the shasum validation fix should run, but the unused permission detection ones shouldn't.

Reply:
Oh, we don't eval, but we still fix the manifest in watch right now. So you are saying we will just skip the permissions validation since we don't do eval?